tests/lib/prepare-uc20.sh

```diff
@@ -0,0 +1,224 @@
+#!/bin/bash
+
+set -e
+set -x 
+
+# these should already be installed in GCE and LXD images with the google/lxd-nested 
+# backend, but in qemu local images from qemu-nested, we might not have them
+if [ "${SPREAD_BACKEND}" = "qemu-nested" ]; then
+    apt update -yqq
+    apt install snapd ovmf qemu-system-x86 sshpass whois -yqq
+fi
+
+snap install snapcraft --classic
+snap install ubuntu-image --classic
+
+# get the model
+curl -o ubuntu-core-20-amd64-dangerous.model https://raw.githubusercontent.com/snapcore/models/master/ubuntu-core-20-amd64-dangerous.model
+
+# download neccessary images
+snap download pc-kernel --channel=20/$SNAP_BRANCH --basename=upstream-pc-kernel
+snap download pc --channel=20/$SNAP_BRANCH --basename=upstream-pc-gadget
+snap download snapd --channel=$SNAP_BRANCH --basename=upstream-snapd
+
+# create test user for spread to use
+groupadd --gid 12345 test
+adduser --uid 12345 --gid 12345 --disabled-password --gecos '' test
+
+if getent group systemd-journal >/dev/null; then
+    usermod -G systemd-journal -a test
+    id test | MATCH systemd-journal
+fi
+
+echo 'test ALL=(ALL) NOPASSWD:ALL' >> /etc/sudoers
+
+# next repack / modify the snaps we use in the image, we do this for a few 
```

Comment:
(nitpick) maybe mention that there is similar/copied code in snapd and that any change/fix in one place should be transferred to the other :)